internal/controller/serviceaccount_controller.go

```diff
@@ -171,17 +149,53 @@ func (r *serviceAccountReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
+func (r *serviceAccountReconciler) provisionSecretForAccount(
+	ctx context.Context, sa *corev1.ServiceAccount, principal string, accountIndex int, totalAccounts int,
```

Comment:
Do "account" and "principal" point the same entity? We should unify the term usage.